capytaine/io/xarray.py

```diff
@@ -300,40 +324,184 @@ def assemble_dataset(results: Sequence[LinearPotentialFlowResult],
+                    """Return the only element of a 1-element dictionnary"""
+                    return next(iter(d.values()))
+                dataset.coords['nb_faces'] = the_only(nb_faces)
+                dataset.coords['quadrature_method'] = the_only(quad_methods)
 
     # HYDROSTATICS
     if hydrostatics:
-        bodies = list({result.body for result in results})
-        dataset = xr.merge([dataset, hydrostatics_dataset(bodies)])
+        if bemio_import:
+            LOG.warning('Bemio data import being used, hydrostatics=True is ignored.')
+        else:
+            bodies = list({result.body for result in results})
+            dataset = xr.merge([dataset, hydrostatics_dataset(bodies)])
 
     dataset.attrs.update(attrs)
     dataset.attrs['capytaine_version'] = __version__
     return dataset
 
 
+#######################
+#  Import from Bemio  #
+#######################
+
+def dataframe_from_bemio(bemio_obj, wavenumber, wavelength):
+    """Transform a :class:`bemio.data_structures.bem.HydrodynamicData` into a
+        :class:`xarray.Dataset`.
+
+        .. todo:: Return bemio attributes as well as a DataFrame
+                  Add rigid body mass
+
+        Parameters
+        ----------
+        bemio_obj: 
+        """
+
+
+    dofs = np.array(['Surge', 'Sway', 'Heave', 'Roll', 'Pitch', 'Yaw'])
+    df = pd.DataFrame()
+    for i in range(bemio_obj.body[0].num_bodies):
+        difr_dict = []
+        rad_dict = []
+
+        rho = bemio_obj.body[0].rho
+        g = bemio_obj.body[0].g
+
+        if bemio_obj.body[i].water_depth == 'infinite':
+            bemio_obj.body[i].water_depth = np.infty
+
+        if bemio_obj.body[i].bem_code == 'WAMIT': # WAMIT coefficients need to be dimensionalized
+            from_wamit = True
+
+        for omega_idx, omega in enumerate(np.sort(bemio_obj.body[i].w)):
+
+            # DiffractionProblem variable equivalents
+            for dir_idx, dir in enumerate(bemio_obj.body[i].wave_dir):
+                temp_dict = {}
+                temp_dict['body_name'] = bemio_obj.body[i].name
+                temp_dict['water_depth'] = bemio_obj.body[i].water_depth
+                temp_dict['omega'] = omega
+                temp_dict['rho'] = rho
+                temp_dict['g'] = g
+                temp_dict['wave_direction'] = np.radians(dir)
+                temp_dict['convention'] = bemio_obj.body[i].bem_code
+                temp_dict['influenced_dof'] = dofs
+                
+                if wavenumber:
+                    if temp_dict['water_depth'] == np.infty or omega**2*temp_dict['water_depth']/temp_dict['g'] > 20:
+                        temp_dict['wavenumber'] = omega**2/temp_dict['g']
+                    else:
+                        temp_dict['wavenumber'] = newton(lambda x: x*np.tanh(x) - omega**2*temp_dict['water_depth']/temp_dict['g'], x0=1.0)/temp_dict['water_depth']
+
+                if wavelength:
+                    if temp_dict['wavenumber'] == 0.0:
```

Comment:
Does this work if `wavenumber=False, wavelength=True`?

Reply:
Thanks for catching this. Fixed in 8a7d81a.